host/src/lib.rs

```diff
@@ -45,5 +49,9 @@ pub fn add_to_linker<T: Send>(
     wasi::exit::add_to_linker(l, f)?;
     wasi::environment::add_to_linker(l, f)?;
     wasi::environment_preopens::add_to_linker(l, f)?;
+    #[cfg(feature = "http")]
+    wasi::default_outgoing_http::add_to_linker(l, f)?;
+    #[cfg(feature = "http")]
+    wasi::http::add_to_linker(l, f)?;
```

Comment:
I believe the `http-types` interface also needs to get added to the linker, and will require a stub implementation, for this to  link against components which use the interface.

Reply:
i renamed this back to the original `types` and put all stub implementation as suggested